src/http/one/MessageParser.js

```diff
@@ -39,14 +41,68 @@ export default class MessageParser {
         }
 
         this.message = new this._messageType();
-        this.message.startLine.parse(match[1]);
-        this.message.header.parse(match[2]);
+        this.parseStartLine(this.message.startLine, match[1]);
```

Comment:
Please refactor so that MessageParser::parseFoo() methods create, parse, and return Foo objects. Set `message.foo` parts to the results of those methods (instead of passing `message.foo` to parsers).